src/dos/drive_local.cpp

```diff
@@ -61,12 +61,13 @@ bool localDrive::FileCreate(DOS_File** file, char* name, FatAttributeFlags attri
 	safe_strcpy(newname, basedir);
 	safe_strcat(newname, name);
 	CROSS_FILENAME(newname);
+	char* expanded_name = dirCache.GetExpandName(newname); //Can only be used in till a new drive_cache action is preformed */
```

Comment:
I recommend to rename this to something that reflects the functionality better. It's tricky because you don't wanna put a novel into the function names, but this is on the too short side that hides the actual action performed by the function.

If I read this name, I have no reason to believe it does case normalisation. The word "expand" in the name gives me a "false sense" of understanding what it does and look no further, but actually this tricks the reader and leads them astray...

Suggest to rename it to `ExpandNameAndNormaliseCase` or something (if that's what it's doing), and also comment the behaviour in its header file. Or just `Normalise` which is so abstract that it will force the reader to look up the documentation of the function.

Reply:
I've renamed it as you suggested.  Also made a copy of the string to not have to worry about the lifetime of the static local pointer that function returns.  I noticed I actually was making another dirCache call while still holding onto that pointer so that could have been a bug.

I had another look at the function and I still don't fully grasp exactly what it does.  It gets passed in a path with all uppercase (I assume that's how the DOS side sees the file since DOS is case insensitive).  It then does a lookup in some cache and eventually returns the path as the host OS sees it.

It's a lot of hard to read C-style string manipulation.  There's also a vector of `CFileInfo` (whatever that is) and a hash function taken from WINE for some reason.

I'd be tempted to rename it to `ManipulateString` or `DoMagic` but your suggestion seems a bit better.

Reply:
Yeah, many times I'm faced with similar decisions and I'm not always prepared to spend days getting to the bottom of things. As long as we're making improvements, every little incremental change that helps cleaning up this mess is welcome 😅